pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -413,18 +422,65 @@ func (h *handler) doRemove(infraObj *infraObject) (runtime.Object, error) {
 	return obj, generic.ErrSkip
 }
 
+func (h *handler) EnqueueAfter(infra *infraObject, duration time.Duration) {
+	err := h.dynamic.EnqueueAfter(infra.obj.GetObjectKind().GroupVersionKind(), infra.meta.GetNamespace(), infra.meta.GetName(), duration)
+	if err != nil {
+		panic(err) // always programmer error
```

Comment:
Why do we want to `panic` here? Is it just to make the issue really obvious in the logs? Other uses of `h.dynamic.EnqueueAfter` in this file handle the error instead of just panicking

Reply:
Opted to log the error instead, its not really any error that can be handled in a meaningful way